forms-flow-web/src/components/FOI/customComponents/Records/index.js

```diff
@@ -1550,437 +2078,532 @@ export const RecordsLog = ({
-            }}
-          >
-            Replace Attachment
-          </MenuItem>}
-          {record.originalfile!=''  && record.originalfile!=undefined  && <MenuItem
-            onClick={() => {
-                handleDownloadoriginal();
-                setPopoverOpen(false);
-            }}
-          >
-            Download Original
-          </MenuItem>
+    const ActionsPopover = ({ RestrictViewInBrowser, record }) => {
+      return (
+        <Popover
+          anchorReference="anchorPosition"
+          anchorPosition={
+            anchorPosition && {
+              top: anchorPosition.top,
+              left: anchorPosition.left,
+            }
           }
-          {((record.isredactionready && conversionFormats.includes(record.attributes?.extension?.toLowerCase())) ||
-          (record.attributes?.isattachment && record.attributes?.trigger === 'recordreplace')) && <MenuItem
-            onClick={() => {
-                handleDownloadPDF();
-                setPopoverOpen(false);
-            }}
-          >
-            {(record.attributes?.isattachment && record.attributes?.trigger === 'recordreplace') ? 'Download Replaced' : 'Download Converted'}
-          </MenuItem>}
-          <MenuItem
-            onClick={() => {
+          open={popoverOpen}
+          anchorOrigin={{
+            vertical: "top",
+            horizontal: "left",
+          }}
+          transformOrigin={{
+            vertical: "top",
+            horizontal: "left",
+          }}
+          onClose={() => setPopoverOpen(false)}
+        >
+          <MenuList>
+            {!RestrictViewInBrowser ? (
+              <MenuItem
+                onClick={() => {
+                  handleView();
+                  setPopoverOpen(false);
+                }}
+              >
+                View
+              </MenuItem>
+            ) : (
+              ""
+            )}
+            {(!record.attributes?.isattachment ||
+              record.attributes?.isattachment === undefined) && (
+              <MenuItem
+                onClick={() => {
+                  handleReplace();
+                  setPopoverOpen(false);
+                }}
+              >
+                Replace Manually
+              </MenuItem>
+            )}
+            {record.attributes?.isattachment && (
+              <MenuItem
+                onClick={() => {
+                  handleReplaceAttachment();
+                  setPopoverOpen(false);
+                }}
+              >
+                Replace Attachment
+              </MenuItem>
+            )}
+            {record.originalfile != "" && record.originalfile != undefined && (
+              <MenuItem
+                onClick={() => {
+                  handleDownloadoriginal();
+                  setPopoverOpen(false);
+                }}
+              >
+                Download Original
+              </MenuItem>
+            )}
+            {((record.isredactionready && record.isconverted) ||
```

Comment:
removed conversionFormats.includes(record.attributes?.extension?.toLowerCase()) and replaced with record.isconverted to download converted file